### ISSUE TITLE
uPlot does not take typed arrays as data

### ISSUE THREAD

**Issue by psarka** [leeoniya/uPlot#717]
This is a comment I got on stack overflow, when my example failed to render: https://stackoverflow.com/questions/72880591/using-uplot-with-svelte-does-not-render/72881761#72881761

I see that it does take typed arrays in the demo (https://github.com/leeoniya/uPlot/blob/master/demos/area-fill.html), but not when used as an npm package. Which is very confusing :thinking: 

Another (simpler) example is to use Vite with vanilla js, and the following files:

index.html
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Vite App</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />

    <link rel="icon" type="image/svg+xml" href="favicon.svg" />
    <link rel="stylesheet" href="node_modules/uplot/dist/uPlot.min.css"/>
  </head>
  <body>
    <div id="app"></div>
    <script type="module" src="/src/main.ts"></script>
  </body>
</html>
```

main.ts
```ts
import uPlot from 'uplot';

let data = [
  new Float32Array([1., 2., 3., 4., 5.]),
  new Float32Array([1., 3., 2., 5., 4.]),
];

const opts = {
  width: 600,
  height: 300,
  scales: {x: {time: false}},
  series: [
      {label: "x"},
      {label: "y", stroke: "red"},
  ],
};

new uPlot(opts, data, document.body);
```

If I remove `Float32Array` things get rendered properly. `npm` installs `uplot` version 1.6.21`.

**Comment by psarka** [comment]
Also, vscode does not like it with red wiggle and message
```
Argument of type 'Float32Array[]' is not assignable to parameter of type '[xValues: number[] | TypedArray, ...yValues: (TypedArray | (number | null | undefined)[])[]]'.
  Source provides no match for required element at position 0 in target.ts(2345)
```

Removing `Float32Array` and leaving plain arrays (which render fine), vscode does not like them with a very similar message
```
Argument of type 'number[][]' is not assignable to parameter of type '[xValues: number[] | TypedArray, ...yValues: (TypedArray | (number | null | undefined)[])[]]'.
  Source provides no match for required element at position 0 in target.ts(2345)
```

**Comment by leeoniya** [comment]
> I see that it does take typed arrays in the demo (https://github.com/leeoniya/uPlot/blob/master/demos/area-fill.html), but not when used as an npm package. Which is very confusing 🤔

typed array fixes were added after 1.6.21 (see #706 and recent commits). the demo uses `master` which is why it works. so it should work in the next published release.

can you try with `"uplot": "leeoniya/uPlot#ddaf5d85a3c796e11868357dd1dac46e9548de61"` in your `package.json` and see if it works?

**Comment by psarka** [comment]
No, it does not work with the proposed version neither.

**Comment by leeoniya** [comment]
hmm, mine is not complaining, maybe some kind of cache on your side or an older TS version? (mine shows 4.7.3)

![image](https://user-images.githubusercontent.com/43234/177624252-81f3d5c9-8a7a-4e7b-ae96-80af7a266df9.png)

**Comment by leeoniya** [comment]
ah, okay. if i omit `AlignedData`, then it complains. i can add TypedArray[] to the signature

**Comment by leeoniya** [comment]
fixed in f0af226e735bafb21feb18755cb95dc1dbdcc7f1
